apps/approver/src/approver.service.ts

```diff
@@ -271,6 +277,91 @@ export class ApproverService {
     }
   }
 
+  async handleWalletUpdated(
+    walletUpdatedData: WalletUpdatedData,
+  ): Promise<boolean> {
+    try {
+      this.logger.info(
+        `[handleWalletUpdated] Received a wallet updated notification, fetching wallet keys data: ${JSON.stringify(
+          walletUpdatedData,
+        )}`,
+      );
+
+      const walletKeys = await this.getWalletKeys(
+        walletUpdatedData.organizationId,
+      );
+
+      if (!walletKeys) {
+        this.logger.error(
+          `[handleWalletUpdated] Failed fetching wallet keys data: ${JSON.stringify(
+            walletUpdatedData,
+          )}`,
+        );
+        return false;
+      }
+
+      await this.walletKeysService.saveWalletKeys(walletKeys);
+      this.logger.error(
+        `[handleWalletUpdated] Successful fetching and saving wallet keys data: ${JSON.stringify(
+          walletUpdatedData,
+        )}`,
+      );
+      return true;
+    } catch (error) {
+      this.logger.error(
+        `[handleWalletUpdated] Failed fetching and saving wallet keys data: ${JSON.stringify(
+          walletUpdatedData,
+        )}`,
+      );
+      return false;
+    }
+  }
+
+  async getWalletKeys(organizationId: string): Promise<string | undefined> {
+    const accessToken = await this.authService.login();
+
+    const apiSigner = await this.authService.getApiSignerState(accessToken);
+
+    if (!apiSigner) {
+      throw new Error('Approver is not registered, contact support.');
+    }
+
+    if (this.authService.pairingNeeded(apiSigner)) {
+      this.logger.info(
```

Comment:
this should be "logger.error".
Consider put the message can be in a const var, because it is the same msg in both log and throw

Reply:
Done